### PR TITLE
Block sync and its child pages

### PR DESCRIPTION
**Pull request by blukat29**
## Proposed changes

Edit and supplement block-sync-related contents throughout the docs.
- Learn
  - (Edit) `storage` Data Management
    - (New) `block-sync` Block Synchronization
    - (Merge) `state-migration` State Migration -> `state-pruning` Storage Optimization
    - (Merge) `live-pruning` StateDB Live Pruning -> `state-pruning` Storage Optimization
- Node Quick Reference
  - (Move) `chaindata-change` Change Chaindata -> `chaindata-snapshot` Use Chaindata Snapshots
  - (Move) `chaindata-migration` Migrate Chaindata -> `node-pruning` Prune Node Data
  - (Edit) `upstream-en` Configure Upstream Archive Node: Upstream EN

For other pages, replace the repeating "Fast Sync" paragraph with a link to `chaindata-snapshot`.
- `nodes/core-cell/install/install-consensus-nodes` 
- `nodes/core-cell/install/install-proxy-nodes`
- `nodes/endpoint-node/docker-setup`
- `nodes/endpoint-node/install-endpoint-nodes`

Pages excluded from the sidebar, link to succeeding pages.
- `/learn/storage/state-migration/`
- `/learn/storage/live-pruning/`
- `/misc/operation/chaindata-change/`
- `/misc/operation/chaindata-migration/`

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Minor Issues and Typos
- [x] Major Content Contribution
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to reach out. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/kaiachain/kaia-docs/blob/main/CONTRIBUTING.md)
- [ ] I have read the [CLA](https://gist.github.com/kaiachain-dev/bbf65cc330275c057463c4c94ce787a6) and signed by comment ```I have read the CLA Document and I hereby sign the CLA``` in first time contribute
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large content contribution, kick off the discussion by explaining why you would suggest the content contribution, etc...

### CODE REVIEW

**Review by hyunsooda — APPROVED** [review]

File: docs/misc/operation/chaindata-migration.md [hunk 1]
```diff
@@ -1,30 +1,3 @@
-# Migrate Chaindata
+# State Migration how to
```
Comment (ayo-klaytn) [inline docs/misc/operation/chaindata-migration.md:1]: Can we reword this to: State Migration: A Step-by-Step Guide ?

File: docs/learn/storage/chaindata-snapshot.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
+# Chaindata snapshot
+
+You can start a node from an already-synced database called a chaindata snapshot. A chaindata snapshot is a compressed Kaia data directory.
+
+ This saves time to [full sync](../block-sync) the whole blockchain, allowing you to relatively quickly start a new node or recover from corrupt database.
+
+## How to
```
Comment (ayo-klaytn) [inline docs/learn/storage/chaindata-snapshot.md:7]: Can we also reword this to: Chaindata Snapshot Setup Process

File: docs/learn/storage/block-sync.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
+#  Block sync
+
+Block sync is a process of following up the latest blocks and states. Node operators can choose from various methods that best fits the machine spec and service requirements.
+
+## Full sync
+
+Full sync generates the block states by downloading every block (header and transactions) and executing it. The blocks are downloaded by p2p peers. A full sync is activated with `--syncmode full` flag or simplly omitting the flag, because full sync is the default sync method.
```
Comment (yoomee1313) [inline docs/learn/storage/block-sync.md:7]:
```suggestion
Full sync generates the block states by downloading every block (header and transactions) from p2p peers and executing it. A full sync is activated with `--syncmode full` flag or simply omitting the flag, because full sync is the default sync method.
```

File: docs/misc/operation/chaindata-migration.md (hunk 1 above)
Comment (blukat29) [reply docs/misc/operation/chaindata-migration.md:1]: This page is merely a redirection. Let me revert to the original title.

File: docs/learn/storage/block-sync.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+Historic block states can be deleted to keep the disk size slim. With the live pruning feature enabled, the block states beyond a certain retention period will be automatically deleted. Read more about live pruning [here](../live-pruning).
+
+### Batch pruning (state migration)
+
+The block states can be state-migrated, meaning the block states before a certain block number is not available. Read more about state migration [here](../state-migration).
+
+## Chaindata snapshot
+
+Download a chaindata snapshot to skip the full sync process and start quickly. A chaindata snapshot is a compressed chaindata directory (e.g. `.tar.gz` file) of an already synced node. Read more about using the chaindata snapshot [here](../)
```
Comment (yoomee1313) [inline docs/learn/storage/block-sync.md:61]: Second sentence of `Snap sync` is more suitable here. Can we move the sentence from `Snap sync` to `Chaindata snapshot`?

File: docs/learn/storage/block-sync.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+## Snap sync
+
+Kaia node currently does not support snap sync. Instead, you can download a state-migrated chaindata snapshot or live-pruning chaindata snapshot to quickly start the node without the need of re-execution from genesis.
```
Comment (yoomee1313) [inline docs/learn/storage/block-sync.md:65]:
```suggestion
Kaia node currently does not support snap sync. Instead, you can download chaindata snapshot.
```

File: docs/learn/storage/block-sync.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+> eth.getBalance('0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266', 150000001)
+Error: missing trie node 64380a8de7bd83a6421c9ad45ae596a0eebbc7b504d061f4a57c61742eadc804 (path )
+	at web3.js:6812:9(39)
+	at send (web3.js:5223:62(29))
+	at <eval>:1:15(4)
+```
+
+#### Usage
+
+It's not a good idea for an application to assume block interval is 128. You might be tempted to rely on the 128 interval because 128 is the default value and most nodes use the default value. But it is the node’s discretion to choose any different interval.
```
Comment (yoomee1313) [inline docs/learn/storage/block-sync.md:41]:
```suggestion
It's not advisable for an application to assume that `state.block-interval` has a default value of 128. While most nodes use the 128 default value, it's not guaranteed. Nodes have the discretion to choose a different interval, so relying on 128 interval could lead to issues.
```

File: docs/learn/storage/block-sync.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+Even if full sync calculates every state transition, the node may not store every block states. By discarding some block states you can significantly save the disk space.
+
+#### Configuration
+
+- Archive mode: Persist every block states. Specify `--gcmode archive` flag.
+- Full mode: Persist some block states. Specify `--gcmode full` flag or simply omit the `--gcmode` flag.
+
+A node that runs on gcmode=archive is called Archive node. A node that runs gcmode=full is called Full node. Not to confuse with the full sync mode.
+
+In a full node, block states are persisted to disk every multiple of the number specified by `--state.block-interval NNN (default: 128)`. Also the block states of the recent `--state.tries-in-memory NNN (default 128)` blocks are kept in memory to serve APIs. Therefore, block states are only available when it is a multiple of block interval or recently processed.
```
Comment (yoomee1313) [inline docs/learn/storage/block-sync.md:24]:
```suggestion
In a full node, block states are persisted to disk every multiple of the number specified by `--state.block-interval NNN (default: 128)`. Also the block states of the recent `--state.tries-in-memory NNN (default: 128)` blocks are kept in memory to serve APIs. Therefore, block states are only available when it is a multiple of block interval or recently processed.
```

File: docs/learn/storage/block-sync.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
+#  Block sync
+
+Block sync is a process of following up the latest blocks and states. Node operators can choose from various methods that best fits the machine spec and service requirements.
+
+## Full sync
```
Comment (yoomee1313) [inline docs/learn/storage/block-sync.md:5]: Can we list up the full sync options here?

File: docs/learn/storage/chaindata-snapshot.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+    mv /var/kend2 /var/kend     # New data
+    ```
+- Option 2. Change the path in the node configuraion
+  - Change `DATA_DIR` value in the `kend.conf` file.
+- Optionally delete old data and tar.gz file.
+- Finally, start the node.
+
+## Downloads
+
+For efficiency, only batch pruned (state migrated) or live pruned database are provided. Read [Block Sync](../block-sync) for their concepts. If you want a full database without neither pruning, or even archive data, consult to [AWS AMIs](../aws-ami) or perform a fresh full sync from genesis.
```
Comment (scott-klaytn) [inline docs/learn/storage/chaindata-snapshot.md:94]: Missing link to "aws-ami.md" page

**Comment by blukat29** [comment]
@scott-klaytn since you're allowed to edit this branch as a Maintainer, feel free to push commits to my branch.

**Comment by blukat29** [comment]
@scott-klaytn This is much clearer and smoothly reading!

**Review by scott-klaytn — APPROVED** [review]

**Review by yoomee1313 — APPROVED** [review]
LGTM
